trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.js

```diff
@@ -83,4 +89,26 @@ describe(component, () => {
       wrapper.expectElement(`${rowSelector} td:contains("${feeRecord.displayStatus}")`).toExist();
     });
   });
+
+  it.each`
+    condition       | feeRecordStatus                     | checkboxShouldExist
+    ${'should'}     | ${FEE_RECORD_STATUS.TO_DO}          | ${true}
+    ${'should'}     | ${FEE_RECORD_STATUS.DOES_NOT_MATCH} | ${true}
+    ${'should not'} | ${FEE_RECORD_STATUS.MATCH}          | ${false}
+    ${'should not'} | ${FEE_RECORD_STATUS.READY_TO_KEY}   | ${false}
+    ${'should not'} | ${FEE_RECORD_STATUS.RECONCILED}     | ${false}
+  `('$condition render the checkbox when the fee record status is $feeRecordStatus', ({ feeRecordStatus, checkboxShouldExist }) => {
+    const feeRecordItem = {
+      ...aFeeRecordItem(),
+      status: feeRecordStatus,
+    };
+    const wrapper = render({ feeRecords: [feeRecordItem] });
+
+    const rowSelector = `[data-cy="premium-payments-table-row--feeRecordId-${feeRecordItem.id}"]`;
+    if (checkboxShouldExist) {
+      wrapper.expectElement(`${rowSelector} input#feeRecordId-${feeRecordItem.id}`).toExist();
+    } else {
+      wrapper.expectElement(`${rowSelector} input#feeRecordId-${feeRecordItem.id}`).notToExist();
+    }
```

Comment:
Could we split this into two tests, one for the 'should' cases and one for the 'should not' cases, to me on the readability vs commonisation scale this feels one step too commonised